### PR TITLE
SAK-42430 Improved and modernized the Calendar design and usability

### PR DESCRIPTION
**Pull request by fostersdesign**
https://jira.sakaiproject.org/browse/SAK-42430

Modernized the look of the Calendar and fixed some usability issues with the interface. Moved inline styles to the stylesheets and moved design from the tables to elements. Cleaned up some repetitive pieces of the code. There are still a lot of places that need to be cleaned up and simplified, so this is just a first pass. 

Before and after screenshots are in SAK-42430.

### CODE REVIEW

File: calendar/calendar-tool/tool/src/java/org/sakaiproject/calendar/tool/CalendarAction.java [hunk 1]
```diff
@@ -3279,10 +3285,27 @@ protected void buildMonthContext(VelocityPortlet portlet,
...
+		String currentView = state.getState();
+		boolean isDefaultView = false;
+		String defaultView = currentPlacement.getPlacementConfig().getProperty( PORTLET_CONFIG_DEFAULT_VIEW);
+		
+		if (StringUtils.compare(defaultView, currentView) == 0)
+		{
+			isDefaultView = true;
+		}
+		
+		return isDefaultView;
```
Comment (plukasew) [inline calendar/calendar-tool/tool/src/java/org/sakaiproject/calendar/tool/CalendarAction.java]: You can just return StringUtils.equals(defaultView, currentView) here

**Comment by bjones86** [comment]
All Codacy complaints were invalid: buttons, hidden inputs and submit inputs do not need or require a corresponding label.

File: calendar/calendar-tool/tool/src/java/org/sakaiproject/calendar/tool/CalendarAction.java (hunk 1 above)
Comment (fostersdesign) [reply calendar/calendar-tool/tool/src/java/org/sakaiproject/calendar/tool/CalendarAction.java]: Thanks for the suggestion, @plukasew!

File: calendar/calendar-tool/tool/src/webapp/vm/calendar/calendar_macros.vm [hunk 2]
```diff
@@ -0,0 +1,37 @@
+## Macros for Calendar ##
+
+## Macro to create the Calendar View picker (to switch between Day, Week, Month, Year, List calendar views)
+#macro(calendarView)
+			<!-- drop down menu for different views -->
+			<form name="viewForm" method="post" action="#toolForm("CalendarAction")">
```
Comment (bjones86) [inline calendar/calendar-tool/tool/src/webapp/vm/calendar/calendar_macros.vm]: Can you reduce the indentation here? The macro you have below has much less rightward drift.

Comment (fostersdesign) [reply calendar/calendar-tool/tool/src/webapp/vm/calendar/calendar_macros.vm]: Sure, I'll take a look. Thanks.

Comment (fostersdesign) [reply calendar/calendar-tool/tool/src/webapp/vm/calendar/calendar_macros.vm]: That indent was based on the HTML markup indenting of where that macro gets inserted, but you're right that it makes it odd and difficult when you are editing that macros file. I'll remove the excess.

Comment (fostersdesign) [reply calendar/calendar-tool/tool/src/webapp/vm/calendar/calendar_macros.vm]: I've removed the indentation.

File: calendar/calendar-tool/tool/src/webapp/vm/calendar/calendar_macros.vm [hunk 3]
```diff
@@ -0,0 +1,40 @@
+## Macros for Calendar ##
+
+## Macro to create the Calendar View picker (to switch between Day, Week, Month, Year, List calendar views)
+#macro(calendarView)
+<!-- drop down menu for different views -->
+<form name="viewForm" method="post" action="#toolForm("CalendarAction")">
+	<div class="sakai-table-viewFilter">
+		<input type="hidden" name="eventSubmit_doView" value="view" />
```
[the remark this answers is not in the document]
Comment (fostersdesign) [reply calendar/calendar-tool/tool/src/webapp/vm/calendar/calendar_macros.vm]: Hidden input. False positive.

File: calendar/calendar-tool/tool/src/webapp/vm/calendar/calendar_macros.vm [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+				#else
+					<option value="$view" >$view</option>
+				#end
+			#end
+		</select>
+		<input type="hidden" name="sakai_csrf_token" value="$sakai_csrf_token" />
+	</form>
+	<form name="defaultViewForm" method="post" action="#toolForm("CalendarAction")">
+		<input type="hidden" name="sakai_csrf_token" value="$sakai_csrf_token" />
+		<input type="submit" name="eventSubmit_doDefaultview" value="$tlang.getString('java.default_view')" #if ($isDefaultView) disabled="disabled" #end />
```
[the remark this answers is not in the document]
Comment (fostersdesign) [reply calendar/calendar-tool/tool/src/webapp/vm/calendar/calendar_macros.vm]: Submit button - no label needed. False positive.

File: calendar/calendar-tool/tool/src/webapp/vm/calendar/calendar_macros.vm [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+					<option value="$view" selected="selected" >$view</option>
+				#else
+					<option value="$view" >$view</option>
+				#end
+			#end
+		</select>
+		<input type="hidden" name="sakai_csrf_token" value="$sakai_csrf_token" />
+	</form>
+	<form name="defaultViewForm" method="post" action="#toolForm("CalendarAction")">
+		<input type="hidden" name="sakai_csrf_token" value="$sakai_csrf_token" />
```
[the remark this answers is not in the document]
Comment (fostersdesign) [reply calendar/calendar-tool/tool/src/webapp/vm/calendar/calendar_macros.vm]: Hidden input: no label needed. False positive.

**Comment by mpellicer** [comment]
Good job @fostersdesign , amazing contribution

**Pull request merged**